daml-assistant/daml-project-config/DAML/Project/Consts.hs

```diff
@@ -137,15 +139,17 @@ getDamlAssistant = getEnv damlAssistantEnvVar
 -- directory into filepaths relative to the project root (absolute file paths will not be modified).
 --
 -- When called outside of a project or outside of the environment setup by the assistant,
--- this function will not modify the current directory.
+-- this function will make the filepath relative to the current working directory
 withProjectRoot :: ((FilePath -> IO FilePath) -> IO a) -> IO a
 withProjectRoot act = do
     previousCwd <- getCurrentDirectory
     mbProjectPath <- getProjectPath
-    case mbProjectPath of
-        Nothing -> act pure
-        Just projectPath -> do
-            projectPath <- canonicalizePath projectPath
-            withCurrentDirectory projectPath $ act $ \f -> do
-                absF <- canonicalizePath (previousCwd </> f)
-                pure (projectPath `makeRelative` absF)
+    let projectPath = fromMaybe previousCwd mbProjectPath
+    projectPath <- canonicalizePath projectPath
+    withCurrentDirectory projectPath $ act $ \f -> do
+        absF <- makeAbsolute (previousCwd </> f)
```

Comment:
Resolving symlinks in `projectPath` but not in `previousCwd </> f` seems like a bad idea. Assuming that there is a reason for this and this is a sane thing to do, we should at least have a comment for this.